src/SFA.DAS.Assessor.Functions/Functions/Ofqual/OfqualStager.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.Assessor.Functions.Data;
+using SFA.DAS.Assessor.Functions.Domain.Entities.Ofqual;
+
+namespace SFA.DAS.Assessor.Functions.Functions.Ofqual
+{
+    public abstract class OfqualStager
+    {
+        private readonly IAssessorServiceRepository _assessorServiceRepository;
+        private readonly OfqualDataType _ofqualDataType;
+        private readonly string _ofqualDataTypeString;
+
+        protected OfqualStager(IAssessorServiceRepository assessorServiceRepository, OfqualDataType ofqualDataType)
+        {
+            _assessorServiceRepository = assessorServiceRepository;
+            _ofqualDataType = ofqualDataType;
+            _ofqualDataTypeString = ofqualDataType.ToString();
+        }
+
+        public async Task<int> InsertDataIntoStagingTable([ActivityTrigger] IEnumerable<IOfqualRecord> records, ILogger logger)
+        {
+            logger.LogInformation("Begin staging of Ofqual {_ofqualDataTypeString} data file.", _ofqualDataTypeString);
+            logger.LogInformation("Checking for downloaded {_ofqualDataTypeString} data file...", _ofqualDataTypeString);
+
+            logger.LogInformation("{records.Count()} records found. Deleting all existing records in {_ofqualDataTypeString} staging table...", records.Count(), _ofqualDataTypeString);
```

Comment:
Presumably this actually works - as in anything in { } is replaced with the variable in the same order - but all of this logging should be done with string interpolation for readability

Reply:
Now removed